project/Admin.cs

```diff
@@ -80,5 +106,17 @@ public Report GenerateReport(string title, JsonArray content)
             );
             return report;
         }
+
+        public void AddTimeslot(Timeslot timeslot)
+        {
+            if (timeslot == null)
+                throw new ArgumentException("Timeslot cannot be null.");
+
+            if (!ManagedTimeslots.Contains(timeslot))
```

Comment:
And what if the timeslot is contained in the data structure? (we need to throw an exception if that is the case here)

Reply:
will add logic to explicitly throw an exception if the timeslot already exists in the ManagedTimeslots list .